src/rgw/rgw_restore.h

```diff
@@ -0,0 +1,142 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab ft=cpp
+
+#pragma once
+
+#include <map>
+#include <array>
+#include <string>
+#include <iostream>
+
+#include "common/debug.h"
+
+#include "include/types.h"
+#include "include/rados/librados.hpp"
+#include "common/ceph_mutex.h"
+#include "common/Cond.h"
+#include "common/iso_8601.h"
+#include "common/Thread.h"
+#include "rgw_common.h"
+#include "cls/rgw/cls_rgw_types.h"
+#include "rgw_sal.h"
+
+#include <atomic>
+#include <tuple>
+
+#define HASH_PRIME 7877
+#define MAX_ID_LEN 255
+static constexpr std::string_view restore_oid_prefix = "restore";
+static constexpr std::string_view restore_index_lock_name = "restore_process";
+
+/** Single Restore entry state */
```

Comment:
I'm not going to be a jerk about this, but I'd generally prefer it if people would put things in the `rgw::` namespace instead of sticking `RGW` at the beginning of identifiers in new code.

Reply:
Done.